src/otx/algo/detection/otx_ssd.py

```diff
@@ -0,0 +1,119 @@
+        config: DictConfig,
+        model_registry: Registry,
+        prefix: str,
+    ) -> dict[str, dict[str, bool | int]]:
+        """Return classification layer names by comparing two different number of classes models.
+
+        Args:
+            config (DictConfig): Config for building model.
+            model_registry (Registry): Registry for building model.
+            prefix (str): Prefix of model param name.
+                Normally it is "model." since OTXModel set it's nn.Module model as self.model
+
+        Return:
+            dict[str, dict[str, int]]
+            A dictionary contain classification layer's name and information.
+            `use_bg` means whether SSD use background class. It if True if SSD use softmax loss, and
+            it is False if SSD use cross entropy loss.
+            `num_anchors` means number of anchors of layer. SSD have classification per each anchor,
+            so we have to update every anchors.
+        """
+        sample_config = deepcopy(config)
+        modify_num_classes(sample_config, 3)
+        sample_model_dict = build_mm_model(sample_config, model_registry, None).state_dict()
+
+        modify_num_classes(sample_config, 4)
+        incremental_model_dict = build_mm_model(sample_config, model_registry, None).state_dict()
+
+        classification_layers = {}
+        for key in sample_model_dict:
+            if sample_model_dict[key].shape != incremental_model_dict[key].shape:
+                sample_model_dim = sample_model_dict[key].shape[0]
+                if sample_model_dim % 3 != 0:
+                    use_bg = True
+                    num_anchors = int(sample_model_dim / 4)
+                    classification_layers[prefix + key] = {"use_bg": use_bg, "num_anchors": num_anchors}
+                else:
+                    use_bg = False
+                    num_anchors = int(sample_model_dim / 3)
+                    classification_layers[prefix + key] = {"use_bg": use_bg, "num_anchors": num_anchors}
+        return classification_layers
+
+    def load_state_dict_pre_hook(self, state_dict: dict[str, torch.Tensor], prefix: str, *args, **kwargs) -> None:
+        """Modify input state_dict according to class name matching before weight loading."""
+        model2ckpt = self.map_class_names(self.model_classes, self.ckpt_classes)
+
+        for param_name, info in self.classification_layers.items():
+            model_param = self.state_dict()[param_name].clone()
+            ckpt_param = state_dict[prefix + param_name]
+            use_bg = info["use_bg"]
+            num_anchors = info["num_anchors"]
+            if use_bg:
+                num_ckpt_classes = len(self.ckpt_classes) + 1
+                num_model_classes = len(self.model_classes) + 1
+            else:
+                num_ckpt_classes = len(self.ckpt_classes)
+                num_model_classes = len(self.model_classes)
+
+            for anchor_idx in range(num_anchors):  # For each anchor
```

Comment:
```suggestion
            for anchor_idx in range(num_anchors):
```
or maybe a more descriptive description?

Reply:
Done by [70ef608](https://github.com/openvinotoolkit/training_extensions/pull/2767/commits/70ef6086ead739796a17faad810b5d47ee5ce06e)